xnmt/input.py

```diff
@@ -26,8 +26,9 @@ class SimpleSentenceInput(Input):
   """
   A simple sent, represented as a list of tokens
   """
```

Comment:
This is actually a very special case for the WordOnlySegmentTransformer where I need to take a look at the external vocab, and so I need to turn back the word ids to its word. Can you explain how to use the Ref(Path())? Or maybe we can just add option to include the original words in the sentence along with the word ids (it will increase the memory use, but will be much simpler).

Reply:
If you need the vocab when you're handling sentences inside the AnnotatedSentenceInput, you could just add a vocab field to```AnnotatedSentenceInput.__init__()```, and then in the config file set it as something like 
```
!WordOnlySegmentTransformer:
  vocab: !Ref { path: model.src_reader.vocab }
  ...
```
(or same with trg_reader if it's on the target side). If that solves the problem, I think that would be cleaner because it doesn't require any changes to the SentenceInput code...

Reply:
This is a good idea! I wonder why didn't I think about this before. Thanks!